alvr/client_openxr/src/extra_extensions/body_tracking_bd.rs

```diff
@@ -0,0 +1,320 @@
+                c"xrLocateBodyJointsBD".as_ptr(),
+                &mut locate_body_joints,
+            );
+
+            locate_body_joints.map(|pfn| mem::transmute::<VoidFunction, LocateBodyJointsBD>(pfn))
+        }
+        .ok_or(sys::Result::ERROR_EXTENSION_NOT_PRESENT)?;
+
+        let start_body_tracking_calib_app = unsafe {
+            let mut start_body_tracking_calib_app = None;
+            let _ = (session.instance().fp().get_instance_proc_addr)(
+                session.instance().as_raw(),
+                c"xrStartBodyTrackingCalibAppBD".as_ptr(),
+                &mut start_body_tracking_calib_app,
+            );
+
+            start_body_tracking_calib_app
+                .map(|pfn| mem::transmute::<VoidFunction, StartBodyTrackingCalibAppBD>(pfn))
+        }
+        .ok_or(sys::Result::ERROR_EXTENSION_NOT_PRESENT)?;
+
+        let get_body_tracking_state = unsafe {
+            let mut get_body_tracking_state = None;
+            let _ = (session.instance().fp().get_instance_proc_addr)(
+                session.instance().as_raw(),
+                c"xrGetBodyTrackingStateBD".as_ptr(),
+                &mut get_body_tracking_state,
+            );
+
+            get_body_tracking_state
+                .map(|pfn| mem::transmute::<VoidFunction, GetBodyTrackingStateBD>(pfn))
+        }
+        .ok_or(sys::Result::ERROR_EXTENSION_NOT_PRESENT)?;
+
+        let props = super::get_props(
+            &session,
+            session
+                .instance()
+                .system(xr::FormFactor::HEAD_MOUNTED_DISPLAY)
+                .unwrap(),
+            SystemBodyTrackingPropertiesBD {
+                ty: *TYPE_SYSTEM_BODY_TRACKING_PROPERTIES_BD,
+                next: ptr::null(),
+                supports_body_tracking: sys::FALSE,
+            },
+        )?;
+
+        if props.supports_body_tracking == sys::FALSE {
+            return Err(sys::Result::ERROR_FEATURE_UNSUPPORTED);
+        }
+
+        let mut handle = XrBodyTrackerBD(0);
+        let info = BodyTrackerCreateInfoBD {
+            ty: *TYPE_BODY_TRACKER_CREATE_INFO_BD,
+            next: ptr::null(),
+            body_joint_set,
+        };
+        unsafe {
+            super::xr_res(create_body_tracker(session.as_raw(), &info, &mut handle))?;
+        };
+
+        let mut status_code = BodyTrackingStatusCodeBD::INVALID;
+        let mut error_code = BodyTrackingErrorCodeBD::INNER_EXCEPTION;
+
+        unsafe {
+            super::xr_res(get_body_tracking_state(
+                session.instance().as_raw(),
+                &mut status_code,
+                &mut error_code,
+            ))?;
+
+            // todo: include actual Android package name
+            let package_name = CString::new("").unwrap().into_raw();
+
+            if status_code == BodyTrackingStatusCodeBD::INVALID
+                || error_code == BodyTrackingErrorCodeBD::TRACKER_NOT_CALIBRATED
+            {
+                super::xr_res(start_body_tracking_calib_app(
```

Comment:
I'm ok for starting body tracking at constructor site, since you are able to just destroy the body tracking handle, so this is not a global state. But also you should check that Pico doesn't have any weird restrictions like Meta about which extensions can be enabled at the same time, for example meta multimodal vs body tracking

Reply:
`start_body_tracking_calib_app` is not a body tracking start. It is just a request to show Motion Tracker application if you need.

Reply:
I see, that's good